sdk/spring-sdk/src/main/java/kalix/springsdk/eventsourced/ReflectiveEventSourcedEntityProvider.java

```diff
@@ -0,0 +1,108 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.eventsourced;
+
+import com.google.protobuf.Descriptors;
+import kalix.javasdk.eventsourcedentity.EventSourcedEntity;
+import kalix.javasdk.eventsourcedentity.EventSourcedEntityContext;
+import kalix.javasdk.eventsourcedentity.EventSourcedEntityOptions;
+import kalix.javasdk.eventsourcedentity.EventSourcedEntityProvider;
+import kalix.javasdk.impl.MessageCodec;
+import kalix.javasdk.impl.eventsourcedentity.EventSourcedEntityRouter;
+import kalix.springsdk.annotations.Entity;
+import kalix.springsdk.impl.ComponentDescriptor;
+import kalix.springsdk.impl.SpringSdkMessageCodec;
+import kalix.springsdk.impl.eventsourcedentity.EventSourcedHandlersExtractor;
+import kalix.springsdk.impl.eventsourcedentity.EventSourceEntityHandlers;
+import kalix.springsdk.impl.eventsourcedentity.ReflectiveEventSourcedEntityRouter;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class ReflectiveEventSourcedEntityProvider<S, E extends EventSourcedEntity<S>>
+    implements EventSourcedEntityProvider<S, E> {
+
+  private final String entityType;
+  private final Function<EventSourcedEntityContext, E> factory;
+  private final EventSourcedEntityOptions options;
+  private final Descriptors.FileDescriptor fileDescriptor;
+  private final Descriptors.ServiceDescriptor serviceDescriptor;
+  private final ComponentDescriptor componentDescriptor;
+
+  private final EventSourceEntityHandlers eventHandlers;
+
+  public static <S, E extends EventSourcedEntity<S>> ReflectiveEventSourcedEntityProvider<S, E> of(
+      Class<E> cls, Function<EventSourcedEntityContext, E> factory) {
+    return new ReflectiveEventSourcedEntityProvider<>(
+        cls, factory, EventSourcedEntityOptions.defaults());
+  }
+
+  public ReflectiveEventSourcedEntityProvider(
+      Class<E> entityClass,
+      Function<EventSourcedEntityContext, E> factory,
+      EventSourcedEntityOptions options) {
+
+    Entity annotation = entityClass.getAnnotation(Entity.class);
+    if (annotation == null)
+      throw new IllegalArgumentException(
+          "Event Sourced Entity [" + entityClass.getName() + "] is missing '@Entity' annotation");
+
+    this.eventHandlers = EventSourcedHandlersExtractor.handlersFrom(entityClass);
+    if (this.eventHandlers.errors().nonEmpty()) {
+      throw new IllegalArgumentException(
+              "Event Sourced Entity [" + entityClass.getName() + "] has event handlers configured incorrectly: " + this.eventHandlers.errors());
```

Comment:
we validate every single possible handler and report all the errors found as opposed to stopping and throwing on first error. Example of what a log looks like:

```json
{"timestamp":"2022-09-02T09:23:23.467Z","thread":"main","logger":"org.springframework.boot.web.reactive.context.AnnotationConfigReactiveWebServerApplicationContext","message":"Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kalixServer' defined in kalix.springsdk.KalixConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [kalix.springsdk.impl.KalixServer]: Factory method 'kalixServer' threw exception; nested exception is java.lang.IllegalArgumentException: Event Sourced Entity [wiring.eventsourcedentity.Counter] has event handlers configured incorrectly: List(ValidationError(reason='must be public, with exactly one parameter and return type 'java.lang.Integer'', offendingMethods=List(notGoodBecausePrivate, wrongReturnType), ValidationError(reason='cannot have duplicate event handlers for the same event type: 'wiring.eventsourcedentity.ValueMultiplied'', offendingMethods=List(handleMultiply, duplicatedEvent))","context":"default","severity":"WARN"}
```